### ISSUE TITLE
Magic number rule does not work as expected

### ISSUE THREAD

**Issue by reisi007** [detekt/detekt#4549]
With default settings (e.g. the number `0` is excluded) the following code (in a method) should not trigger the rule:

```var test : Long = 0```

As a workaround, the following code works:

```var test = 0L```

As both of these are equivalent, I expect that none of those code samples trigger the rule....

**Comment by cortinico** [comment]
> With default settings (e.g. the number `0` is excluded) the following code (in a method) should not trigger the rule:

I've just tested this and I'm unable to reproduce. I verified that the code you specified is not reported by the `MagicNumber` rule. I suspect there is something going on with your local configuration.

Would you be able to provide a reproducer?
